### PR TITLE
Feature/thousand separators

### PR DESCRIPTION
**Pull request by gergelyvagujhelyi**
- Display numbers with comma thousand separators (e.g. 1,234,567) in expressions, result preview, history label, and history entries                                                                                                                                                                                                                                  
  - Rewrite formatExpression from regex to character-by-character parser that handles both operator spacing and separator insertion
  - Rewrite cursor mapping functions (mapCursorToFormatted / mapCursorFromFormatted) to account for separator characters so tap-to-position cursor remains accurate                                                                                                                                                                                                     
  - Add formatResultNumber for standalone result/preview formatting                                                                                                                                                                                                                                                                                                     
  - Add 26 unit tests covering separator placement, cursor round-trips, E notation, negatives, and edge cases                                                                                                                                                                                                                                                           
                                                                                                                                                                                                                                                                                                                                                                        
  Details                                                                                                                                                                                                                                                                                                                                                               
                                                                                                                                                                                                                                                                                                                                                                        
  - Separators are display-only — raw expressions and evaluation are unaffected                                                                                                                                                                                                                                                                                         
  - Only the integer part of numbers gets separators; decimal digits and E notation are left untouched
  - Numbers with 3 or fewer digits are unchanged

### CODE REVIEW

File: app/src/main/java/com/vagujhelyigergely/calculatorm3/CalculatorScreen.kt [hunk 1]
```diff
@@ -910,36 +910,143 @@ fun HistorySheet(
...
 
-private fun mapCursorToFormatted(raw: String, rawCursor: Int): Int {
-    var formattedPos = 0
-    for (i in 0 until rawCursor.coerceAtMost(raw.length)) {
-        if (raw[i] in listOf('+', '−', '×', '÷')) {
-            formattedPos += 3 // " X "
-        } else {
-            formattedPos += 1
+internal fun formatResultNumber(value: String): String {
+    if (value.isEmpty() || value.startsWith("Error") || value.contains('E') || value.contains('e')) return value
```
[the remark this answers is not in the document]
Comment (gergelyvagujhelyi) [reply app/src/main/java/com/vagujhelyigergely/calculatorm3/CalculatorScreen.kt:972]: The bot noticed formatResultNumber skips E notation while formatExpression would format the integer part of one. In practice this never manifests: E notation comes from formatForDisplay which uses engineering notation (1-3 digits before E), so the integer part is always too short to get separators anyway. Not a bug.

File: app/src/main/java/com/vagujhelyigergely/calculatorm3/CalculatorScreen.kt [hunk 2]
```diff
@@ -910,36 +910,143 @@ fun HistorySheet(
...
+                    }
+                    while (i < expr.length && expr[i].isDigit()) {
+                        sb.append(expr[i])
+                        i++
+                    }
+                }
+            }
+            // Binary operators: space around them (not at position 0).
+            // ASCII '-' (from +/− toggle) is unary, excluded from this set.
+            c in setOf('+', '×', '÷', '−') && sb.isNotEmpty() -> {
```
[the remark this answers is not in the document]
Comment (gergelyvagujhelyi) [reply app/src/main/java/com/vagujhelyigergely/calculatorm3/CalculatorScreen.kt]: 12d589cef18d8a807e1ea32acc1dceb5acdd5f69

File: app/src/main/java/com/vagujhelyigergely/calculatorm3/CalculatorScreen.kt [hunk 3]
```diff
@@ -910,36 +910,143 @@ fun HistorySheet(
...
+    val abs = if (negative) value.substring(1) else value
+    val dotIndex = abs.indexOf('.')
+    val intPart = if (dotIndex >= 0) abs.substring(0, dotIndex) else abs
+    if (intPart.length <= 3) return value
+    val rest = if (dotIndex >= 0) abs.substring(dotIndex) else ""
+    val formatted = buildString {
+        for (j in intPart.indices) {
+            if (j > 0 && (intPart.length - j) % 3 == 0) append(',')
+            append(intPart[j])
         }
```
[the remark this answers is not in the document]
Comment (gergelyvagujhelyi) [reply app/src/main/java/com/vagujhelyigergely/calculatorm3/CalculatorScreen.kt:983]: 12d589cef18d8a807e1ea32acc1dceb5acdd5f69
